server/cli/src/cli.rs

```diff
@@ -396,6 +405,102 @@ async fn main() -> anyhow::Result<()> {
+                    .collect::<Result<Vec<String>, std::io::Error>>()?;
+
+                let (_, name) = name_dir.rsplit_once('/').unwrap();
+
+                for version_dir in report_versions {
+                    let (_, version) = version_dir.rsplit_once('/').unwrap();
+
+                    let args = BuildArgs {
+                        dir: format!("{version_dir}/src"),
+                        output: format!("{version_dir}/generated/{name}.json").into(),
+                        template: "template.html".to_string(),
+                        header: None,
+                        footer: None,
+                        query_gql: Some("query.graphql".to_string()),
+                        query_default: None,
+                        query_sql: None,
+                    };
+
+                    let _ = build(args);
+
+                    let arguments_path = format!("{version_dir}/argument_schemas/arguments.json");
+                    let arguments_ui_path =
+                        format!("{version_dir}/argument_schemas/arguments_ui.json");
+
+                    let manifest_file = fs::File::open(format!("{version_dir}/manifest.json"))
+                        .expect("file should open read only");
+
+                    let manifest: Manifest = serde_json::from_reader(manifest_file)
+                        .expect("manifest json not formatted");
+
+                    let id = format!("{name}_{version}");
+                    let report_path = format!("{version_dir}/generated/{name}.json");
+                    let context = manifest.context;
+                    let report_name = manifest.name;
+                    let is_custom = manifest.is_custom;
+
+                    let sub_context = manifest.sub_context;
+                    let code = manifest.code;
+
+                    let filter = ReportFilter::new().id(EqualFilter::equal_to(&id));
+                    let existing_report =
+                        ReportRepository::new(&con).query_by_filter(filter)?.pop();
+
+                    let argument_schema_id = existing_report
+                        .and_then(|r| r.argument_schema.as_ref().map(|r| r.id.clone()));
+
+                    let form_schema_json = schema_from_row(FormSchemaRow {
+                        id: argument_schema_id.unwrap_or(format!("for_report_{}", id)),
+                        r#type: "reportArgument".to_string(),
+                        json_schema: fs::read_to_string(arguments_path)?,
+                        ui_schema: fs::read_to_string(arguments_ui_path)?,
+                    })?;
+
+                    FormSchemaRowRepository::new(&con).upsert_one(&form_schema_json)?;
+
+                    ReportRowRepository::new(&con).upsert_one(&ReportRow {
+                        id,
+                        name: report_name,
+                        r#type: repository::ReportType::OmSupply,
+                        template: fs::read_to_string(report_path)?,
```

Comment:
Would be nice if we can just return the string from teh report builder rather than creating the .json file?
Can re-look at that when we do the embedding stuff I think.